src/rfc3986/iri.py

```diff
@@ -41,11 +40,17 @@ class IRIReference(
         the future. Check for changes to the interface when upgrading.
     """
 
-    slots = ()
```

Comment:
Again, I didn't recall why this might be necessary but I feel like it is. Does this need to be removed to fix typing?

Reply:
No that makes sense too. I suspect this may be a Py2 left over